samples/datasync-server-cosmosdb-singlecontainer/src/Program.cs

```diff
@@ -0,0 +1,86 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using CommunityToolkit.Datasync.Server;
+using CommunityToolkit.Datasync.Server.Abstractions.Json;
+using CommunityToolkit.Datasync.Server.CosmosDb;
+using CommunityToolkit.Datasync.Server.Swashbuckle;
+using Microsoft.Azure.Cosmos;
+using Sample.Datasync.Server.SingleContainer.Models;
+using System.Collections.ObjectModel;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+
+string connectionString = builder.Configuration.GetConnectionString("DefaultConnection")
+    ?? throw new ApplicationException("DefaultConnection is not set");
+
+CosmosClient cosmosClient = new CosmosClient(connectionString,
+    new CosmosClientOptions()
+    {
+        UseSystemTextJsonSerializerWithOptions = new()
+        {
+            Converters =
+            {
+                new JsonStringEnumConverter(),
+                new DateTimeOffsetConverter(),
+                new DateTimeConverter(),
+                new TimeOnlyConverter(),
+                new SpatialGeoJsonConverter()
+            },
+            PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+            DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull | JsonIgnoreCondition.WhenWritingDefault,
+            NumberHandling = JsonNumberHandling.AllowNamedFloatingPointLiterals,
+            ReferenceHandler = ReferenceHandler.Preserve
+        }
+    });
+
+builder.Services.AddSingleton(cosmosClient);
+builder.Services.AddSingleton<ICosmosTableOptions<TodoItem>>(new CosmosSharedTableOptions<TodoItem>("TodoDb", "TodoContainer"));
+builder.Services.AddSingleton<ICosmosTableOptions<TodoList>>(new CosmosSharedTableOptions<TodoList>("TodoDb", "TodoContainer"));
+builder.Services.AddSingleton(typeof(IRepository<>), typeof(CosmosTableRepository<>));
```

Comment:
Is there significant overhead with creating a repository per request?  Why not just create the repository within the table controller?

Ditto for the table options - why are they done here and not in the table controller?

Reply:
I believe this is be correct use of DI. I like my config in one place not throughout the code, If I was going to swap out the implementation I'm doing it then in one place. I know this is complicated slightly by the provider specific ITableData implementation in this project. I've opted for Singleton registration in this case as there's no per request dependencies in the Repository or the Options, the CosmosClient is also singleton which is recommended. 

Changing the registration to scoped would move these to be resolved per request.